components/activities/activities-tab/activities-tab.tsx

```diff
@@ -118,20 +93,54 @@ export function ActivitiesTabManager<
         <div className="list-header">
           ACTIVITY HISTORY
         </div>
-        {pastActivities?.length
-          ? (
-            <div className="list-group">
-              {pastActivities?.map((activity) => {
-                const ActivityTile = activity.sourceType
-                  ? activityListItems[activity.sourceType]
-                  : null;
-                return ActivityTile &&
-                  React.createElement(ActivityTile, { activity: activity.activity });
-              })}
-            </div>
-          )
-          : <div className="empty-list-group">No past activities</div>}
+        {/* activity li's will only rerender if their data changes */}
+        <Memo>
+          {() => (
+            <ActivityGroup
+              activityConnection$={activityAlertConnection$}
+              activityListItems={activityListItems}
+              isActive={false}
+              emptyStateMessage="No past activities"
+            />
+          )}
+        </Memo>
       </div>
     </div>
   );
 }
+
+function ActivityGroup<
+  ActivityListItemTypes,
+  SourceType extends StringKeys<ActivityListItemTypes>,
+  ActivityType extends ActivityListItemTypes[SourceType],
+>(
+  { activityConnection$, activityListItems, isActive, emptyStateMessage }: {
+    activityConnection$: Observable<ActivityConnection<ActivityType, SourceType>>;
+    activityListItems: ListItemMap<ActivityListItemTypes>;
+    isActive: boolean;
+    emptyStateMessage: string;
+  },
+) {
+  const activities = useSelector(() =>
+    activityConnection$?.nodes.get()?.filter((activity) =>
+      activity?.activity && isActive ? isActiveActivity(activity) : !isActiveActivity(activity)
+    )
+  );
+
+  return activities?.length
+    ? (
+      <div className="list-group">
+        {activities?.map((activity) => {
+          const ActivityListItem = activity.sourceType
+            ? activityListItems[activity.sourceType]
+            : null;
+
+          return ActivityListItem &&
+            React.createElement(ActivityListItem, {
+              activity: activity.activity,
```

Comment:
don't need to solve for now, but in future we should try to avoid properties on the model that are the same name as the model. makes for confusing bits like this where it's unclear if an activity is an obj or a str

Reply:
yeah, technically should just rename the variable since the model is Alert. so should follow alert.activity or activityAlert.activity

Reply:
I can update to clear up confusion